include/extractor/node_based_graph_factory.hpp

```diff
@@ -41,7 +41,7 @@ class NodeBasedGraphFactory
                           std::vector<TurnRestriction> &turn_restrictions,
                           std::vector<ConditionalTurnRestriction> &conditional_turn_restrictions);
 
-    auto const &GetGraph() const { return compressed_output_graph; }
+    auto &GetGraph() { return compressed_output_graph; }
```

Comment:
This should remain const, to not break consistency. From what I understand you should be fine adding this back in, after now operating on the compressed graph.

Reply:
Need to have access to non-const EdgeData to set segregated flag.

Reply:
I kind of hate to have non-const access to the graph. This implies that we can do further modifications, and the graph itself should remain the same. Can we potentially do a Decorator pattern here to only add the data to the EdgeBasedNode container, instead of modifying the node-based graph?

Reply:
If you do want to keep it on the edges, the method should probably be called from within the node-based graph factory, to ensure that the graph is in a consistent state after the production step.

Reply:
OK, I can build the vector of segregated edge ids and pass it to the BuildEdgeExpandedGraph -> EdgeBasedGraphFactory. In this case we avoid segregated flag in node-based graph edges.

Reply:
I am on Moritz's side to keep the node-based graph immutable to prevent possible problems with data races. Using a vector of segregated edges is fine for me. The guidance code has to be refactored later within scope of #4664